sdks/python/apache_beam/runners/portability/local_job_service.py

```diff
@@ -102,42 +98,13 @@ def Cancel(self, request, context=None):
 
   def GetStateStream(self, request, context=None):
     job = self._jobs[request.job_id]
-    state_queue = queue.Queue()
-    job.add_state_change_callback(state_queue.put)
-    try:
-      current_state = state_queue.get()
-    except queue.Empty:
-      current_state = job.state
-    yield beam_job_api_pb2.GetJobStateResponse(state=current_state)
-    while current_state not in TERMINAL_STATES:
-      current_state = state_queue.get(block=True)
-      yield beam_job_api_pb2.GetJobStateResponse(state=current_state)
+    for state in job.GetStateStream():
+      yield state
```

Comment:
I think the logic of turning a state into a GetJobStateResponse logically belongs here, not in BeamJob. Similarly for messages.

Reply:
Done